### PR TITLE
use com.optimaize.langdetect for lang detection

### PR DESCRIPTION
**Pull request by jefimm**
https://github.com/vespa-engine/vespa/issues/6403

### CODE REVIEW

**Review by bratseth — APPROVED** [review]

**Comment by gjoranv** [comment]
Please hold merge until I've added a commit to fix the dependencies. The purpose of the enforcer is to prevent adding any dependencies that are not provided at runtime to the maven tree of external plugin developers.

File: container-dependencies-enforcer/pom.xml [hunk 1]
```diff
@@ -136,6 +136,10 @@
                                                 <include>org.slf4j:slf4j-api:[${slf4j.version}]:jar:provided</include>
                                                 <include>org.slf4j:slf4j-jdk14:[${slf4j.version}]:jar:provided</include>
                                                 <include>xml-apis:xml-apis:[1.4.01]:jar:provided</include>
+                                                <include>com.optimaize.languagedetector:language-detector:0.6:jar:provided</include>
```
Comment (gjoranv) [inline container-dependencies-enforcer/pom.xml]: Please remove all changes in dependencies-enforcer pom.
See #6460

**Review by gjoranv — CHANGES_REQUESTED** [review]

**Comment by jefimm** [comment]
@gjoranv  done

**Comment by gjoranv** [comment]
The container-dev module is only a helper pom for plugin development. When developing a searcher plugin, users should add a dependency to the `container` maven artifact, which includes container-dev. The exclusions are done to "hide" the apis that are _not_ exported runtime from the osgi container, to force users to add them explicitly in their own pom. Otherwise, you could have a working build, but a runtime failure because the classes are not available.

The libraries referred to here will be available for the linguistics module inside the container, but not for code outside the bundle that contains the linguistics code.

**Comment by jefimm** [comment]
@gjoranv  Could you please advise how to fix the failing tests ?

**Comment by gjoranv** [comment]
I think I see what the problem is. Our internal pom files generate the wrong test classpath. I will try to get it fixed tonight or tomorrow morning outside this PR.

**Review by gjoranv — APPROVED** [review]

**Pull request merged**